pkg/platform/kube/functionres/lazy.go

```diff
@@ -2184,23 +2191,68 @@ func (lc *lazyClient) resolveFailFast(pods []v1.Pod,
 				}
 			}
 		}
+	}
 
+	errGroup, _ := errgroup.WithContext(context.TODO(), lc.logger)
+	for _, pod := range pods {
+		pod := pod
 		for _, condition := range pod.Status.Conditions {
 
 			// check if the pod is in pending state, and the reason is that it is unschedulable
 			// (meaning no k8s node can currently run it, because of insufficient resources etc..)
 			if pod.Status.Phase == v1.PodPending && condition.Reason == "Unschedulable" {
 
-				// TODO: if pods has an event that cluster is going to scale up - don't return error
-
-				return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+				errGroup.Go("WaitAndCheckAutoScaleEvents", func() error {
+					time.Sleep(15 * time.Second)
+
+					triggeredScaleUp, err := lc.isPodAutoScaledUp(pod, namespace)
+					if err != nil {
+						if triggeredScaleUp {
+							return errors.Wrap(err, "Failed to get pod events")
+						}
+						return errors.Wrapf(err,
+							"Autoscaler is unable to schedule NuclioFunction pod (%s)",
+							pod.Name)
+					}
+					return nil
```

Comment:
This entire block should be written in a less complexed way.

```golang
triggeredScaleUp, err := lc.isPodAutoScaledUp(pod, namespace)
if err != nil {
    return errors.Wrapf(err,
        "Failed to resolve pod (%s) triggered a node scale up",
        pod.Name)
}
if triggeredScaleUp {
        // log something that a cluster scale up just happened...
        // communicate back that scale up occurred. you can use sync.Lock a change a value from multiple writers. a boolean whould be sufficient.
       // later on, callee of that err group will wait for scale up? increase timeout? etc

      
}
return nil
```

Reply:
If we don't return an error, then we keep waiting for deployment to be ready.
We were talking about increasing the overall timeout, regardless of scaling up (maybe in a different PR).
Isn't that the case?